synapse/http/matrixfederationclient.py

```diff
@@ -714,6 +720,26 @@ async def _send_request(
                             response.code,
                             response_phrase,
                         )
+                    elif (
+                        response.code in (307, 308)
```

Comment:
I thought about wrapping the agent in a `RedirectAgent`, but that handles additional codes sadly.